### PR TITLE
fix: fix handling of ordered arrays when processing them async

### PR DESCRIPTION
**Pull request by TristanSpeakEasy**

### CODE REVIEW

File: datamodel/high/base/schema.go
Comment (bitomaxsp) [inline datamodel/high/base/schema.go:344]: Leftover?

Comment (TristanSpeakEasy) [reply datamodel/high/base/schema.go]: yep good catch thanks

**Comment by bitomaxsp** [comment]
Neat! thanks for fixing it this way. I can close #59 then

**Comment by TristanSpeakEasy** [comment]
> @TristanSpeakEasy When do you plan to merge this ?

As soon as I get an approval from @daveshanley

**Review by daveshanley — APPROVED** [review]
This looks great, an excellent solution, without losing the async speed! Thank you for your contribution.

**Pull request merged**
